lib/compress/index.js

```diff
@@ -1598,6 +1598,53 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    let default_or_exact = default_branch || exact_match;
+
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every((branch, i) =>
+        (branch === default_or_exact || !branch.expression.has_side_effects(compressor))
+        && (branch.body.length === 0 || branch.body.some(aborts) || body.length - 1 === i))
+    ) {
+        for (let i = 0; i < body.length; i++) {
+            const branch = body[i];
+            if (branch.body.length === 0) continue;
+            if (!aborts(branch)) continue;
+            for (let j = i + 1; j < body.length; j++) {
+                const next = body[j];
+                if (next.body.length === 0) continue;
+                const last_branch = j === (body.length - 1);
+                const equivalentBranch = branches_equivalent(next, branch, false);
+                if (equivalentBranch || (last_branch && branches_equivalent(next, branch, true))
+                ) {
+                    if (!equivalentBranch && last_branch) {
+                        if (!is_break(next.body[next.body.length - 1])) {
+                            next.body.push(make_node(AST_Break));
+                        }
```

Comment:
We already know this is true by the `!equivalentBranch && last_branch` check.
```suggestion
                    if (!equivalentBranch && last_branch) {
                        next.body.push(make_node(AST_Break));
```

Reply:
correct